frontend/js/pages/BattleDetail.js

```diff
@@ -0,0 +1,100 @@
+import { get, map } from 'lodash';
+import React, { useState } from 'react';
+import { useParams } from 'react-router-dom';
+import styled from 'styled-components';
+
+import { apiUrls, getFromApi } from 'utils/api';
+
+const Title = styled.h1`
+  font-size: 1.5em;
+  text-align: center;
+  color: white;
+`;
+
+const Wrapper = styled.section`
+  padding: 4em;
+  background: linear-gradient(to right, rgb(197, 230, 236), rgb(239, 187, 230));
+  text-align: center;
+  align-items: center;
+`;
+
+function BattleDetail() {
+  const { id } = useParams();
+  const [battle, setBattle] = useState();
+  const currentUrl = window.location.host;
+  const url = `http://${currentUrl}/battle/list/`;
```

Comment:
Details of urls should not be in your component, remember the single responsibility concept